src/views/Validator/ValidatorControls.tsx

```diff
@@ -1,31 +1,46 @@
 import { Button, DropdownMenu, IconButton, Tooltip } from '@radix-ui/themes'
 import { EllipsisVerticalIcon } from 'lucide-react'
 
+import { DeleteFileDialog } from '@/components/DeleteFileDialog'
 import TextSpinner from '@/components/TextSpinner/TextSpinner'
 import { GrafanaIcon } from '@/components/icons/GrafanaIcon'
+import { useDeleteFile } from '@/hooks/useDeleteFile'
 import { useProxyStatus } from '@/hooks/useProxyStatus'
+import { StudioFile } from '@/types'
 
 interface ValidatorControlsProps {
+  file: StudioFile
   isRunning: boolean
   canDelete: boolean
-  onDeleteScript: () => void
   onRunScript: () => void
   onRunInCloud: () => void
   onSelectScript: () => void
   onStopScript: () => void
+  onAfterDelete: () => void
 }
 
 export function ValidatorControls({
+  file,
   isRunning,
   canDelete,
-  onDeleteScript,
   onRunScript,
   onRunInCloud,
   onSelectScript,
   onStopScript,
+  onAfterDelete,
 }: ValidatorControlsProps) {
   const proxyStatus = useProxyStatus()
 
+  const handleDelete = useDeleteFile({
+    file,
+    navigateHomeOnDelete: false,
```

Comment:
You are disabling navigation here, but pass a `onAfterDelete` prop that does exactly that. Perhaps `onAfterDelete` could be removed?